src/SystemCommands-RefactoringSupport/RBPullUpPreview.class.st

```diff
@@ -0,0 +1,90 @@
+"
+I am a preview of Pull up refactoring, I can select the superClass of pull up methos, and select the methods of class to pull up
+"
+Class {
+	#name : #RBPullUpPreview,
+	#superclass : #RBPushDownPreview,
+	#instVars : [
+		'dropList',
+		'label2'
+	],
+	#category : #'SystemCommands-RefactoringSupport'
+}
+
+{ #category : #specs }
+RBPullUpPreview class >> defaultSpec [
+	^ SpBoxLayout newVertical
+		add: #label2 withConstraints: [ :c | c height: self labelHeight ];
+		add:
+			(SpBoxLayout newHorizontal
+				add: #dropList ;
+				yourself) withConstraints: [ :c | c height: self toolbarHeight  ];
+		add: #label withConstraints: [ :c | c height: self buttonHeight ];
+		add:
+			(SpBoxLayout newHorizontal
+				add: #table;
+				yourself);
+		yourself
+]
+
+{ #category : #examples }
+RBPullUpPreview class >> example [
+	<script>
+	^ (self on: RBPullUpPreview withSelectors: Array new)
+]
+
+{ #category : #specs }
+RBPullUpPreview class >> title [
+
+	^ 'Pull members up to'
+]
+
+{ #category : #'instance creation' }
+RBPullUpPreview >> accept [
+	refactoring superClass: self activeClass .
+	refactoring selectors: (selectedMethods collect: #selector).
+]
+
+{ #category : #updating }
+RBPullUpPreview >> activeClass [
+	^dropList selectedItem
```

Comment:
Question about Spec2. Does drop list allow to set the target value holder?
I would expect here to see a #targetClass value (with targetClass as a value holder)